src/lib/types/array.js

```diff
@@ -83,7 +83,7 @@ function arrayType(value, path, resolve, traverseCallback) {
   const optionalsProbability = optionAPI('alwaysFakeOptionals') === true ? 1.0 : optionAPI('optionalsProbability');
   const fixedProbabilities = optionAPI('alwaysFakeOptionals') || optionAPI('fixedProbabilities') || false;
 
-  let length = random.number(minItems, maxItems, 1, 5);
```

Comment:
Why this change?

Reply:
This ensures that if `alwaysFakeOptionals` is set, the array length is always at least 1 -- if a user set `minItems` as 0 but also set `alwaysFakeOptionals`, this would preference `alwaysFakeOptionals`. Does that make more sense,  as I'm putting it?

Reply:
Not sure, if you revert this change, then the tests you modified are not passing due this?

Reply:
After reading your feedback, I realized that I *can* revert the tests I changed(!) -- I've done so just now in 15f69208. However, without this change, one test fails:

```
  203 passing (14s)
  1 failing

  1) will use alwaysFakeOptionals (tests/schema/core/issues/issue-494.json)
       should work if enabled:
     Error: account-response.current should not be empty
```

With this change in place, all tests pass for me.

Reply:
Now I see, these lines are affecting the whole thing:

https://github.com/json-schema-faker/json-schema-faker/pull/543/files#diff-46bf7bc97cebdac6a541eb79a0ed3cd6R61-R62

The original logic below is meant to fix that kind of stuff, related with minItems and maxItems, so could you review such thing?

Reply:
> so could you review such thing

Sure, can do. I'll revise the approach, and re-comment here once it's ready for another look.